### DISCUSSION TITLE
Sample clicks and pops on note off

### DISCUSSION THREAD

**Discussion by randorec** [monocasual/giada#672] [Ideas]
Hi all! I'm happy to see that this project is still being developed. Thank you for your dedication.

I've come across a common issue when playing/triggering samples - clicks and pops. This usually occurs when a sample is started or stopped at a non-zero crossing point. In my case I was triggering a bass sample and upon note off, there's an audible pop. The full sample is properly trimmed with the start and end at zero crossings. However the note length for the tune is variable, so a note off command can occur at any point in the sample.

The usual fix for this is to apply an envelope with a short attack and fast decay to zero upon a note off command. Envelopes are not part of the sample editor or sample channel yet. My suggestion is to add this or something similar to prevent pops.

I've been looking for a workaround, but I haven't found anything yet. I might assign that part to a MIDI instrument, but I'll loose the feel of the sample by doing that.

Once again, thank you for continuing to improve this application. What you've done here to date is remarkable.

**Comment by gvnnz** [comment]
Hey @randorec, thanks a lot for your kind words. Yes, Giada is being actively developed, we are about to release the glorious version 1.0 in a few weeks and there are tons of cool features in our pipeline. Speaking of which, issue https://github.com/monocasual/giada/issues/656 is about applying an envelope with short attack as you suggested to avoid nasty clicks. It will be implemented between version 1.0 and 2.0, in the next few months. Stay tuned!
